From: Anita Zhang <the.anitazha@gmail.com>
Date: Sun, 12 Sep 2021 00:02:31 -0700
Subject: test: tweak parameters for TEST-55-OOMD

Pressure remains > 1% after a kill for some time and could cause
testchill to get killed. Bumping the limit from 1% to 20% should help
with this.

Fixes #20118
---
 test/units/testsuite-55-workload.slice | 2 +-
 test/units/testsuite-55.sh             | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/test/units/testsuite-55-workload.slice b/test/units/testsuite-55-workload.slice
index 8c32b28..a1accbc 100644
--- a/test/units/testsuite-55-workload.slice
+++ b/test/units/testsuite-55-workload.slice
@@ -7,4 +7,4 @@ MemoryAccounting=true
 IOAccounting=true
 TasksAccounting=true
 ManagedOOMMemoryPressure=kill
-ManagedOOMMemoryPressureLimit=1%
+ManagedOOMMemoryPressureLimit=20%
diff --git a/test/units/testsuite-55.sh b/test/units/testsuite-55.sh
index 099ecc9..e988dc5 100755
--- a/test/units/testsuite-55.sh
+++ b/test/units/testsuite-55.sh
@@ -38,7 +38,7 @@ systemctl start testsuite-55-testbloat.service
 
 # Verify systemd-oomd is monitoring the expected units
 oomctl | grep "/testsuite-55-workload.slice"
-oomctl | grep "1.00%"
+oomctl | grep "20.00%"
 oomctl | grep "Default Memory Pressure Duration: 2s"
 
 systemctl status testsuite-55-testchill.service
@@ -75,7 +75,7 @@ if setfattr -n user.xattr_test -v 1 /sys/fs/cgroup/; then
         if ! systemctl status testsuite-55-testmunch.service; then
             break
         fi
-        sleep 5
+        sleep 2
     done
 
     # testmunch should be killed since testbloat had the avoid xattr on it
